File: spring-data/spring-data-r2dbc/src/main/java/com/gmail/unmacaque/spring/data/r2dbc/domain/Message.java
package com.gmail.unmacaque.spring.data.r2dbc.domain;

import org.springframework.data.annotation.Id;

public class Message {

	@Id
	private Long id;

	private String title;

	private String author;

	private String content;

	public String getContent() {
		return content;
	}

	public Long getId() {
		return id;
	}

	public String getAuthor() {
		return author;
	}

	public String getTitle() {
		return title;
	}

	public void setContent(String content) {
		this.content = content;
	}

	public void setId(long id) {
		this.id = id;
	}

	public void setAuthor(String author) {
		this.author = author;
	}

	public void setTitle(String title) {
		this.title = title;
	}
}

File: spring-kotlin/src/main/kotlin/com/gmail/unmacaque/spring/domain/Message.kt
package com.gmail.unmacaque.spring.domain

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
data class Message(
    val sender: String,
    val recipient: String,
    val content: String,
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) val id: Long = -1
)
